frontend/libs/studio-components/src/components/StudioSearch/StudioSearch.tsx

```diff
@@ -0,0 +1,24 @@
+import React, { forwardRef } from 'react';
+import { Label, Search, type SearchProps } from '@digdir/designsystemet-react';
+import type { WithoutAsChild } from '../../types/WithoutAsChild';
+import { StudioFieldset } from '../StudioFieldset';
+
+export type StudioSearchProps = WithoutAsChild<SearchProps> & { legend?: string };
+
+const StudioSearch = forwardRef<HTMLInputElement, StudioSearchProps>(
+  ({ size = 'sm', label, legend, ...rest }, ref) => {
+    const showLabel = !!label;
+    return showLabel ? (
+      <StudioFieldset legend={legend}>
+        <Label>{label}</Label>
+        <Search {...rest} size={size} ref={ref} />
```

Comment:
Når vi bruker `label`, må vi bruke `htmlFor` til å referere til riktig felt. Dette refererer til [HTML-attributtet `for`](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/for). ID-en kan vi generere ved hjelp av [`useId`](https://react.dev/reference/react/useId).
```suggestion
        <Label htmlFor={id}>{label}</Label>
        <Search {...rest} id={id} size={size} ref={ref} />
```
Dette kan vi verifisere med en test som bruker `screen.getByRole('searchbox', { name: label })` til å finne søkefeltet.

For øvrig lurer jeg på hva som er hensikten med `StudioFieldset` her?

Reply:
Skal fikse!

StudiofieldSet la jeg til fordi DS hadde dette kodeeksempelet bare. Men jeg er på tynn is her, så godt mulig det ikke er riktig! https://storybook.designsystemet.no/iframe.html?viewMode=docs&id=komponenter-search--docs&globals=#med-label

Reply:
I dette eksemplet bruker de `Field`, ikke `Fieldset`. Jeg går ut fra at dette er noe som er lagt til versjon 1 for å fikse dette med ID automatisk.

Reply:
Mulig. Men okei, da fjerner jeg det.